rules/0132.go

```diff
@@ -0,0 +1,195 @@
+						Message:    fmt.Sprintf("method %q has no `parent` field", m.Name()),
+						Descriptor: m.Input(),
+					})
+					return problems, nil
+				}
+
+				// Rule check: Establish that the name field is a string.
+				if parentField.Kind() != p.StringKind {
+					problems = append(problems, lint.Problem{
+						Message:    "`parent` field on List RPCs should be a string",
+						Descriptor: parentField,
+					})
+				}
+
+				return problems, nil
+			},
+		},
+	}
+}
+
+// Get methods should not have unrecognized fields.
+func checkListRequestMessageUnknownFields() lint.Rule {
+	return &descriptor.CallbackRule{
+		RuleInfo: lint.RuleInfo{
+			Name:         lint.NewRuleName("core", "0132", "request_message", "unknown_fields"),
+			Description:  "check that there are no unknown fields",
+			RequestTypes: []lint.RequestType{lint.ProtoRequest},
+			URI:          "https://aip.dev/131#request-message",
+		},
+		Callback: descriptor.Callbacks{
+			MethodCallback: func(m p.MethodDescriptor, s lint.DescriptorSource) (problems []lint.Problem, err error) {
+				// We only care about List- methods for the purpose of this rule;
+				// ignore everything else.
+				if !strings.HasPrefix(string(m.Name()), "List") {
+					return
+				}
+
+				// Rule check: Establish that there are no unexpected fields.
+				allowedFields := []string{
+					"parent", "page_size", "page_token",
+					"filter", "order_by", "group_by", "show_deleted",
+				}
+				fields := m.Input().Fields()
+				for i := 0; i < fields.Len(); i++ {
+					field := fields.Get(i)
+					allowed := false
+					for _, allowedField := range allowedFields {
+						if string(field.Name()) == allowedField {
+							allowed = true
+						}
+					}
+					if !allowed {
+						problems = append(problems, lint.Problem{
+							Message:    "List RPCs should only contain fields explicitly described in AIPs.",
+							Descriptor: field,
+						})
+					}
+				}
+
+				return problems, nil
+			},
+		},
+	}
+}
+
+// Get messages should use the resource as the respose message
+func checkListResponseMessageName() lint.Rule {
+	return &descriptor.CallbackRule{
+		RuleInfo: lint.RuleInfo{
+			Name:         lint.NewRuleName("core", "0132", "response_message", "name"),
+			Description:  "check that List RPCs have appropriate response messages",
+			RequestTypes: []lint.RequestType{lint.ProtoRequest},
+			URI:          "https://aip.dev/131#guidance",
+		},
+		Callback: descriptor.Callbacks{
+			MethodCallback: func(m p.MethodDescriptor, s lint.DescriptorSource) (problems []lint.Problem, err error) {
+				// We only care about List- methods for the purpose of this rule;
+				// ignore everything else.
+				methodName := string(m.Name())
+				if !strings.HasPrefix(methodName, "List") {
+					return
+				}
+
+				// Rule check: Establish that for methods such as `ListFoos`, the response
+				// message is named `ListFoosResponse`.
+				responseMessageName := string(m.Output().Name())
```

Comment:
Suggest encapsulating this into a re-usable function. The requirement that the response match the method name + response is common to Create, List, and all custom methods.  Don't love this copy&paste...

Reply:
Done. :)